File: server-v1/src/db/models/comment.model.js
const { Model, DataTypes, Sequelize } = require("sequelize");
const COMMENT_TABLE = "comment";

const CommentSchema = {
  id: {
    allowNull: false,
    autoIncrement: true,
    primaryKey: true,
    type: DataTypes.INTEGER,
  },
  content: {
    allowNull: false,
    type: DataTypes.TEXT,
  },
  createdAt: {
    allowNull: false,
    type: DataTypes.DATE,
    defaultValue: Sequelize.NOW,
  },
};

class Comment extends Model {
  static associate(models) {
    // Relación 1:N entre Post y Comment
    this.belongsTo(models.Post, {
      foreignKey: "postId",
      as: "post",
    });
    // Relación 1:N entre User y Comment
    this.belongsTo(models.User, {
      foreignKey: "userId",
      as: "user",
    });
  }
  static config(sequelize) {
    return {
      sequelize,
      tableName: COMMENT_TABLE,
      modelName: "Comment",
      timestamps: false,
    };
  }
}

module.exports = { COMMENT_TABLE, CommentSchema, Comment };

File: server-v2/src/models/comment_model.py
from datetime import datetime
import config.db as db

db = db.db

class Comment(db.Model):
  id = db.Column(db.Integer, primary_key=True, autoincrement=True)
  content = db.Column(db.Text, nullable=False)
  created_at = db.Column(db.DateTime, nullable=False, default=datetime.utcnow)
  # Relación con la tabla User
  user_id = db.Column(db.Integer, db.ForeignKey('user.id'), nullable=False)
  # Relación con la tabla Post
  post_id = db.Column(db.Integer, db.ForeignKey('post.id'), nullable=False)

  def __init__(self, content, user_id, post_id):
    self.content = content
    self.user_id = user_id
    self.post_id = post_id
